app/ui-react/syndesis/cypress/integration/setup/01-create.setup.spec.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * 01: CREATE DATA
+ *
+ * The purpose of this "test" is to setup data as the user would in the UI.
+ * First, the database is reset, and then custom Cypress commands are used
+ * to create the data manually.
+ */
+
+const constants = require('../../fixtures/constants');
+
+describe('Create Data', () => {
```

Comment:
Is this the right way to do initialization for tests?  Is this a one-time thing or an every test thing?

I get kinda suspicious of tests that need to run in a particular order.

Reply:
The reason I separated these setup tasks into separate tests is for our own convenience. For instance, I could throw all of these into a single setup test, but then that would mean that each and every time we run that test we are resetting the DB, creating the data, and taking a snapshot. I figured that there may be a time that someone would want only a snapshot of the DB, or maybe only to reset the DB, and so on.

The reason they are named in a specific order with numbers in the name is because Cypress runs tests alphabetically, and it wouldn't make sense to create a snapshot and _then_ create the data, for instance. If it were all in one test, this probably wouldn't be an issue because we set up the test manually, but of course means everything would be bundled into one.

I can put these all into a single test if you'd like. Likewise, I can create an npm command for each of these tasks/tests and then the naming and order really wouldn't matter. However, running `yarn e2e:setup` and then clicking 'Run All Tests', would run them in alphabetical order, which wouldn't work.

Last thing I thought I'd mention is that the reason these are written as tests is because the environment Cypress provides is the best way to use the Cypress command, execute node commands, and perform API requests.